File: app/src/main/java/org/potados/study_in_java/MainViewModel.java
package org.potados.study_in_java;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class MainViewModel {

    private String msg = "Hello, World!";

    private MutableLiveData<String> _message = new MutableLiveData<String>(msg);
    public LiveData<String> message = _message;

    public void onButtonClick() {
        _message.postValue(_message.getValue() + msg);
    }
}

File: workmanagerpractice/src/main/java/org/potados/workmanagerpractice/MainViewModel.kt
package org.potados.workmanagerpractice

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainViewModel : ViewModel() {
    private val _progress = MutableLiveData<String>()
    val progress: LiveData<String> = _progress

    fun setProgress(progress: Int) {
        _progress.postValue("Currently at $progress%")
    }
}